src/main/java/edu/kis/powp/jobs2d/features/DrawOnFreePanelFeature.java

```diff
@@ -0,0 +1,15 @@
+package edu.kis.powp.jobs2d.features;
+
+import edu.kis.powp.appbase.Application;
+import edu.kis.powp.jobs2d.drivers.DriverManager;
+
+public class DrawOnFreePanelFeature {
+
+	private static Application app;
+
+	public static void setupButtonClick(Application application, DriverManager driverManager) {
+		app = application;
+
+		app.getFreePanel().addMouseListener(new MouseInputAdapterForSetupButtonClick(app.getFreePanel(),driverManager.getCurrentDriver()));
```

Comment:
app.getFreePanel()....(app.getFreePanel()) it looks to me like some recursive dependency and it is difficult for programmer to come up with. I suggest introduce something like static method :
MouseInputAdapterForSetupButtonClick.enable(app.getFreePanel(), driverManager) that would encapsulate this a bit awkward code.